src/os/windows/win32_main.cpp

```diff
@@ -76,7 +74,7 @@ int APIENTRY WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLi
 	SetRandomSeed(GetTickCount());
 
 	char *argv[64]; // max 64 command line arguments
-	int argc = ParseCommandLine(cmdline, argv, lengthof(argv));
+	int argc = ParseCommandLine(cmdline.data(), argv, lengthof(argv));
```

Comment:
Maybe `ParseCommandLine()` could be modified to work with `std::string` directly.

Reply:
It definitely could be, but... then you rather want to do it properly. Like returning a vector of strings. That would mean changing `openttd_main` to accept that vector instead, which means `getopt` code needs to be made string-aware and that could potentially cause changes in strgen/settingsgen.
That's something you rather want to tackle from the other side, so first make `getopt` string aware, then a vector of strings to `openttd_main` and then rewriting this function.

That's quite a significant thing to do, so I rather first finish the string parameters stuff which already is part of the eradicate stredup "project". The string parameters are waiting on reviews, before I can continue further, so I'm tackling the remaining uses of `stredup` besides the string parameters in quite minimalistic style.